biz.aQute.bndlib/src/aQute/bnd/osgi/Processor.java

```diff
@@ -2216,14 +2218,16 @@ public static ScheduledExecutorService getScheduledExecutor() {
 	 */
 	public synchronized void addBasicPlugin(Object plugin) {
 		basicPlugins.add(plugin);
-		if (plugins != null)
-			plugins.add(plugin);
+		Set<Object> p = plugins;
```

Comment:
Just a hygiene habit. I copy the mutable instance variable into a local variable and then use it knowing the local variable cannot be altered during this method's execution.